functions/index.js

```diff
@@ -1,174 +1,224 @@
+    // Getting details of the new document
+    const appointmentId = context.params.appointmentId;
+    const newAppointmentData = snap.data();
+    const clientDocID = newAppointmentData.clientDocID;
+    if ( isClient(newAppointmentData.creator) ) {
+       console.log('Appointment created by client. need to notify the admin');
+       const clientName = newAppointmentData.clientName ?? 'User';
+       const title = clientName + ' Created new appointment';
+       const services = newAppointmentData.services ? newAppointmentData.services.length : 0;
+       const msg = clientName + ' created new appointment for ' +  services + ' services'
+       // Setting notification content - must send package_id
+       const notificationContent = {
+           notification: {
+               title: title,
+               body: msg,
+               sound: 'default'
+           },
+           data: {
+               client_id: clientDocID,
+               appointment_id: appointmentId,
+               category: 'NotificationCategory.appointment',
+           }
+       };
+       const adminTokens = await getAdminTokens();
+       return admin.messaging().sendToDevice(adminTokens, notificationContent);
+    }
+    if ( isBusiness(newAppointmentData.creator) ) {
+        console.log('Appointment created by business. need to notify the client');
+        const clientTokens = await getClientTokens(clientDocID);
+        if(clientTokens) {
+             const title = 'Aleen Created new appointment for you';
+             const services = newAppointmentData.services ? newAppointmentData.services.length : 0;
+             const msg = 'New appointment with ' +  services + ' services'
+             const notificationContent = {
+                notification: {
+                  title: title,
+                  body: msg,
+                  sound: 'default'
+                },
+                data: {
+                  client_id: clientDocID,
+                  appointment_id: appointmentId,
+                  category: 'NotificationCategory.appointment',
+                }
+            };
+            console.log('Sending notification to ', newAppointmentData.clientName)
+            admin.messaging().sendToDevice(clientTokens, notificationContent);
+        }
     }
+}
 
-    // Exit if the image uploaded is not part of profile.
-    if (filePath.startsWith('profile/')) {
-        return profileImagesHandler(object);
+async function HandleAppointmentUpdate(change, context) {
+    // Get the updated object representing the updated document
+    const newValue = change.after.data();
+    const appointmentId = change.after.id;
+    // Get the previous object before this updated document
+    const previousValue = change.before.data();
+    // check the editor
+    const changedByClient = isClient(newValue.lastEditor);
+    const changedByBusiness = isBusiness(newValue.lastEditor);
+    const clientDocID = newValue.clientDocID;
```

Comment:
Currently, Admin can change client in update screen, hence, you will send update message to "new" client selected

I think we should strict the behavior here, client CAN NOT be changed changed, the expected flow should be:
1. cancel old appointment
2. create new one with new client